java/ql/src/semmle/code/java/dataflow/internal/TaintTrackingUtil.qll

```diff
@@ -51,10 +51,11 @@ predicate localAdditionalTaintStep(DataFlow::Node src, DataFlow::Node sink) {
   localAdditionalTaintUpdateStep(src.asExpr(),
     sink.(DataFlow::PostUpdateNode).getPreUpdateNode().asExpr())
   or
-  exists(Argument arg |
-    src.asExpr() = arg and
-    arg.isVararg() and
-    sink.(DataFlow::ImplicitVarargsArray).getCall() = arg.getCall()
+  exists(Type elemType |
+    arrayReadStep(src, sink, elemType) and
+    not elemType instanceof PrimitiveType and
```

Comment:
Comment on why these exclusions are made here?

Reply:
Because numbers usually don't carry taint. I.e. by default we'll say that an `int[]` can be tainted without implying that the individual elements are tainted, whereas when a `MyObject[]` is tainted then we'll take that to mean that all of the contained `MyObject`s are tainted as well.

Reply:
You could say that we're adding a taint step here rather than restricting one, since the general `x` to `x[0]` step is now considered an array read step that's expected to be matched by an array store step.